recipes/dynare_preprocessor/build.sh

```diff
@@ -0,0 +1,6 @@
+#!/bin/bash
+
+export BOOST_ROOT=$PREFIX
+meson setup build_preproc -Dcpp_link_args='-pthread -static'
```

Comment:
> I made a static build.

We strongly prefer that packages are dynamically linked on our channel in order to avoid the same packages/code being shipped, compiled, stored, and downloaded multiples times. If you need static linking of non-header-only packages, please provide justification.

Reply:
It didn't work before I tried to add the static option ;-) On my computer it was still linking to some of my personal libraries outside of conda.
Let me try again without the static argument.

Reply:
@carterbox : the build passes the online test.
If I try to install the artifact from CI and install locally it also work. But I get:
```
ldd $(which dynare-preprocessor)                                                                                                                                                                                        (base) 
        linux-vdso.so.1 (0x00007ffcc87ed000)
        libstdc++.so.6 => /home/pablo/micromamba/bin/../lib/libstdc++.so.6 (0x00007f662fc1d000)
        libm.so.6 => /lib/x86_64-linux-gnu/libm.so.6 (0x00007f6630243000)
        libgcc_s.so.1 => /home/pablo/micromamba/bin/../lib/libgcc_s.so.1 (0x00007f6630228000)
        libpthread.so.0 => /lib/x86_64-linux-gnu/libpthread.so.0 (0x00007f6630223000)
        libc.so.6 => /lib/x86_64-linux-gnu/libc.so.6 (0x00007f662fa00000)
        librt.so.1 => /lib/x86_64-linux-gnu/librt.so.1 (0x00007f663021c000)
        /lib64/ld-linux-x86-64.so.2 (0x00007f6630353000)
```
Is that ok ?
It is not totally clear to me how much of the shared libraries must be provided by conda, vs expected to be present on a reasonable system. Can you enlighten me here?
On the other hand the static build is really small in size ;-)

Reply:
The Dynare preprocessor is only linked against standard C and C++ libraries. I’m not familiar with Conda, but I would be surprised if those were not shipped as shared library by the package system.

Reply:
Thank you @carterbox for looking into it. Now I understand well. I'll wait for @svillemot for confirmation that we shouldn't use libm or openlibm and remove the deps